tree/treeplayer/inc/ROOT/TInterface.hxx

```diff
@@ -1065,12 +1065,12 @@ protected:
          unsigned int nSlots = df->GetNSlots();
          TBufferMerger merger(filename.c_str(), "RECREATE");
          std::vector<std::shared_ptr<TBufferMergerFile>> files(nSlots);
-         std::vector<std::unique_ptr<TTree>> trees(nSlots);
+         std::vector<TTree*> trees(nSlots);
 
          auto fillTree = [&merger, &trees, &files, &bnames, &treename](unsigned int slot, Args &... args) {
             if (!trees[slot]) {
                files[slot] = merger.GetFile();
-               trees[slot].reset(new TTree(treename.c_str(), treename.c_str()));
+               trees[slot] = new TTree(treename.c_str(), treename.c_str());
```

Comment:
How certain are you which file this tree is associated with? (I.e. should you `merger.GetFule()->cd()` first?)